jib-core/src/main/java/com/google/cloud/tools/jib/api/DockerDaemonImage.java

```diff
@@ -70,16 +73,17 @@ public DockerDaemonImage setDockerExecutable(Path dockerExecutable) {
     return this;
   }
 
-  /**
-   * Converts into an {@link ImageConfiguration}. For internal use only.
-   *
-   * @return an {@link ImageConfiguration}
-   */
   @Override
   public ImageConfiguration toImageConfiguration() {
     return ImageConfiguration.builder(imageReference).build();
   }
 
+  @Override
+  public BuildSteps toBuildSteps(BuildConfiguration buildConfiguration) {
```

Comment:
This feels a little weird. I would expect image doesn't need to know about the BuildSteps class?

Reply:
The `BuildSteps` only vary based on the image target (we should probably rename the interfaces to `ImageTarget` and `ImageSource`) - and generating the `BuildSteps` in `JibContainerBuilder`, for example, would involve logic around checking for being instance of `DockerDaemonImage`.